ground/gcs/src/plugins/boards_matek/matek405.cpp

```diff
@@ -0,0 +1,257 @@
+ * @file       matek405.cpp
+ * @author     Tau Labs, http://taulabs.org, Copyright (C) 2013
+ * @author     dRonin, http://dronin.org Copyright (C) 2015
+ *
+ * @addtogroup GCSPlugins GCS Plugins
+ * @{
+ * @addtogroup Boards_MatekPlugin Matek boards support Plugin
+ * @{
+ * @brief Plugin to support Matek boards
+ *****************************************************************************/
+
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, see <http://www.gnu.org/licenses/>
+ */
+
+#include "matek405.h"
+
+#include "uavobjects/uavobjectmanager.h"
+#include "uavobjectutil/uavobjectutilmanager.h"
+#include <extensionsystem/pluginmanager.h>
+#include "board_usb_ids.h"
+
+#include "hwmatek405.h"
+
+/**
+ * @brief MATEK405::MATEK405
+ *  This is the MATEK405 board definition
+ */
+MATEK405::MATEK405(void)
+{
+    // Common USB IDs
+    addBootloaderUSBInfo(
+        USBInfo(DRONIN_VID_DRONIN_BOOTLOADER, DRONIN_PID_DRONIN_BOOTLOADER, BCD_DEVICE_BOOTLOADER));
+    addFirmwareUSBInfo(
+        USBInfo(DRONIN_VID_DRONIN_FIRMWARE, DRONIN_PID_DRONIN_FIRMWARE, BCD_DEVICE_FIRMWARE));
+    
+    boardType = 0x95;
+
+    // Define the bank of channels that are connected to a given timer
+    channelBanks.resize(5);
+    channelBanks[0] = QVector<int>() << 1;
+    channelBanks[1] = QVector<int>() << 2<< 3 << 4;
+	channelBanks[2] = QVector<int>() << 5;
+	channelBanks[3] = QVector<int>() << 6;
+	channelBanks[4] = QVector<int>() << 7;
+}
+
+MATEK405::~MATEK405()
+{
+}
+
+QString MATEK405::shortName()
+{
+    return QString("MATEK405");
+}
+
+QString MATEK405::boardDescription()
+{
+    return QString("The MATEK405 board");
+}
+
+//! Return which capabilities this board has
+bool MATEK405::queryCapabilities(BoardCapabilities capability)
+{
+    switch (capability) {
+    case BOARD_CAPABILITIES_GYROS:
+    case BOARD_CAPABILITIES_ACCELS:
+    case BOARD_CAPABILITIES_MAGS:
+    case BOARD_CAPABILITIES_BAROS:
+    case BOARD_CAPABILITIES_UPGRADEABLE:
+        return true;
+    default:
+        break;
+    }
+
+    return false;
+}
+
+QPixmap MATEK405::getBoardPicture()
+{
+    return QPixmap(":/matek/images/matek405.png");
+}
+
+//! Determine if this board supports configuring the receiver
+bool MATEK405::isInputConfigurationSupported(Core::IBoardType::InputType type)
+{
+    switch (type) {
+    case INPUT_TYPE_PWM:
+    case INPUT_TYPE_UNKNOWN:
+        return false;
+    default:
```

Comment:
I always preferred to have no default case here, so whenever a new protocol is added, it's impossible to forget about a board (the build will fail... although maybe not anymore since #2243). What do you think?

Reply:
Not a bad idea, but most, if not all, of the targets have the default case here, FWIW.